src/fpm.f90

```diff
@@ -41,7 +41,7 @@ subroutine build_model(model, settings, package, error)
     integer :: i, j
     type(package_config_t) :: dependency
     character(len=:), allocatable :: manifest, lib_dir, flags, cflags, cxxflags, ldflags
-    logical :: has_cpp
+    logical :: has_cpp = .false.
```

Comment:
Please don't introduce implicitly saved variables here, this is generally considered a bug. Initialization needs to go in a separate line, the one below also seems incorrect to me.

Reply:
Why is that so? I'd say that it is safer to declare and initialize variables at the same time because you won't run into the situation where a variable is declared but not initialized. I also found an "standard-conforming F90 code" example that is doing the same:

https://gcc.gnu.org/onlinedocs/gfortran/Old-style-variable-initialization.html

Reply:
Ok, I found out why:

https://stefanoborini.com/fortran-90-pitfall-initialization-of-vars-at-declaration/

This is really terrible, honestly.

Reply:
> This is really terrible, honestly.

I [agree](https://github.com/fortran-lang/fpm/issues/359).